### PR TITLE
Fix funny generics in request size limit test

### PR DESCRIPTION
**Pull request by nik9000**

### CODE REVIEW

**Comment by nik9000** [comment]
@danielmitterdorfer I was trying to debug a non-reproducable failure I had on this test locally (no node available while doing after test cleanup) and I found some funky stuff going on with the generics. Would you mind having a look?

File: core/src/test/java/org/elasticsearch/http/netty/NettyHttpRequestSizeLimitIT.java [hunk 1]
```diff
@@ -79,11 +80,11 @@ public void testLimitsInFlightRequests() throws Exception {
             ().boundAddresses());
 
         try (NettyHttpClient nettyHttpClient = new NettyHttpClient()) {
+            @SuppressWarnings("unchecked")
```
Comment (danielmitterdorfer) [inline core/src/test/java/org/elasticsearch/http/netty/NettyHttpRequestSizeLimitIT.java]: I think the better option is to add the '@SafeVarargs` annotation to `NettyHttpClient#post()` (which requires the method to be `final` but I'd prefer to make even the class `final` as it is not intended for inheritance).

**Comment by danielmitterdorfer** [comment]
@nik9000 I wonder why the problem appeared in the first place. Do you use Eclipse? I did not have any problem in IDEA and on the command line with and Oracle JDK 1.8.0_74-b02.

I left one comment, otherwise LGTM.

Can you please backport this to 2.x too as I just pushed the backport of #17133 where this change comes from?

**Comment by nik9000** [comment]
> I wonder why the problem appeared in the first place. Do you use Eclipse?

Yes I do, but Eclipse didn't catch it. I add type parameters to things that are missing them when I touch code out of habit and that caused a chain reaction. I suspect the that I hit is something you've already fixed by raising the request buffer but it never reproduced locally for me.

**Comment by nik9000** [comment]
2.x: 9486bb04052f11a07e848fcba2c659672ba2e631 08bd55b7d4b946a04d0262e69f77d0788df37646

**Pull request merged**